crates/storage/traits/src/storage.rs

```diff
@@ -44,6 +44,19 @@ where
     }
 }
 
+/// Decode and consume all of the given input data.
+///
+/// If not all data is consumed, an error is returned.
+pub fn decode_all<T: Storable>(input: &mut &[u8]) -> Result<T, scale::Error> {
+    let res = <T as Storable>::decode(input)?;
+
+    if input.is_empty() {
+        Ok(res)
+    } else {
+        Err("Input buffer has still data left after decoding!".into())
```

Comment:
Perhaps would be useful to give the length of the remaining bytes left in a buffer for debugging purposes

Reply:
I'm not sure it is worth bringing in all the string formatting machinery in for that...we need to be mindful of code size. 

It's possible to debug this without this information by fetching the data at the given key and attempting to decode it into the respective type.